test/SILOptimizer/access_enforcement_noescape.swift

```diff
@@ -372,15 +339,16 @@ func readBoxWriteInout() {
 // CHECK-LABEL: } // end sil function '_T027access_enforcement_noescape17readBoxWriteInoutyyFyycfU_'
 
 // Error: inout cannot be captured.
-// func inoutReadBoxWriteInout(x: inout Int) {
-//   let c = { _ = x }
-//   doOneInout(c, &x)
-// }
+// Verification disabled because is suppresses other errors.
+//func inoutReadBoxWriteInout(x: inout Int) {
+//  disabled-error@+1{{escaping closures can only capture inout parameters explicitly by value}}
+//  let c = { _ = x }
```

Comment:
To answer your first question. The idea behind this file is to cover the entire space involving closure reads/writes along with `inout` arguments on both the caller and callee side. The tests are ordered such that it's easy to demonstrate full coverage. Of course, some of these cases are static errors. I tried mixing -verify and FileCheck patterns to handle both swift 3 and 4. Unfortunately, some compile-time errors prevent later diagnostics from running and reporting the other errors!